hnsecure/src/main/java/ca/bc/gov/hlth/hnsecure/parsing/PopulateReqHeader.java

```diff
@@ -36,8 +38,11 @@ public void populateReqHeader(Exchange exchange, @Headers Map<String, Object> hm
 		final String methodName = Util.getMethodName();
 		String recApp = Util.getReceivingApp(v2Message);
 		String msgType = Util.getMsgType(v2Message);
+		String accessToken = (String) exchange.getIn().getHeader(AUTHORIZATION);
+		String sendingFacility = Util.getSendingFacility(accessToken);
```

Comment:
The access token should already be in the exchange header and you're also parsing the Facility in the  Request/ResponseFileDropGenerator classes. You probably don't need to do it in both locations

Reply:
This is the only way to make it working for messages destined to different end points. Access token get modified so need to save it in header before messages route to their end point.
Incase of validation errors, headers are not set , so in that it has to be parsed and retrieved in filedrops